core/runtime/src/commonMain/kotlin/com/kitakkun/backintime/core/runtime/internal/EventReportUtils.kt

```diff
@@ -38,18 +40,18 @@ internal fun reportMethodInvocation(
 )
 
 @BackInTimeCompilerInternalApi
-internal fun reportPropertyValueChange(
+internal inline fun <reified T> reportPropertyValueChange(
     instance: BackInTimeDebuggable,
     ownerClassFqName: String,
     methodInvocationId: String,
     propertyFqName: String,
-    propertyValue: Any?,
+    propertyValue: T,
 ) = getBackInTimeDebugService().processInstanceEvent(
     BackInTimeDebuggableInstanceEvent.PropertyValueChange(
         instance = instance,
         methodCallId = methodInvocationId,
         propertyName = propertyFqName,
-        propertyValue = propertyValue,
+        propertyValue = backInTimeJson.encodeToString(propertyValue),
```

Comment:
This line might throw serialization exception. We need to catch them and report to the debugger in a proper way.

Reply:
[Safely catch and report serialization error](https://github.com/kitakkun/back-in-time-plugin/pull/142/commits/c834237f2eeffef2a878e540e717d8c8ff93714e)